api/src/main/java/org/dicekeys/api/DerivationRecipe.kt

```diff
@@ -85,19 +133,26 @@ data class DerivationRecipe constructor(
                 return null
             }
 
-            val allowDomainList = domains.map { """{"host":"*.$it"}""" }
-            val name = domains.joinToString(", ")
-            var recipeJson = """{"allow":[${allowDomainList.joinToString(",")}]}"""
-
-            if (type == DerivationOptions.Type.Password) {
-                recipeJson = addLengthInCharsToDerivationOptionsJson(recipeJson, lengthInChars)
-            }else if (type == DerivationOptions.Type.Secret) {
-                recipeJson = addLengthInBytesToDerivationOptionsJson(recipeJson, lengthInBytes)
+            val jsonObject = buildJsonObject {
```

Comment:
What's going on here?  I believe the only time we should be adding a *. to a domain is when the user pastes a URL into the recipe builder.